app/src/main/java/space/taran/arkmemo/data/repositories/VersionStorage.kt

```diff
@@ -0,0 +1,139 @@
+package space.taran.arkmemo.data.repositories
+
+import space.taran.arkmemo.data.VersionMeta
+import space.taran.arkmemo.files.parsers.JsonParser
+import space.taran.arkmemo.models.Version
+import space.taran.arkmemo.space.taran.arkmemo.utils.CODES_DELETING_NOTE
+import java.io.*
+import java.nio.file.Files
+import java.nio.file.Path
+import kotlin.io.path.createDirectories
+
+typealias ResourceId = Long
```

Comment:
`ResourceId` from the library should be used to avoid type diverging in future: https://github.com/ARK-Builders/arklib-android/blob/3334850384f84fe8d5f0545a221cc1fbc14e722e/lib/src/main/java/space/taran/arklib/Lib.kt